geode-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisService.java

```diff
@@ -42,6 +45,13 @@ public void close() {
     stopRedisServer();
   }
 
+  @Override
+  public void handleEvent(ResourceEvent event, Object resource) {
+    if (event.equals(ResourceEvent.CACHE_SERVICE_CREATE) && resource == this) {
+      startRedisServer(cache);
+    }
```

Comment:
Is it possible that the `CACHE_SERVICE_CREATE` event is already triggered before we call the `init` method, and then we would just miss it entirely?

Should the `stopRedisServer` be invoked in a similar way, i.e. by listening to the cache lifecycle events?

Reply:
I've added a different event now.